src/Forms/GridField/GridField.php

```diff
@@ -444,6 +451,72 @@ private function initState(): void
         }
     }
 
+    /**
+     * Adds state for this gridfield from the request variables.
+     *
+     * If there is state already set on this GridField, that takes precedent
+     * over state from the request.
+     */
+    private function addStateFromRequest(): void
+    {
+        $request = $this->getRequest();
+        if (($request instanceof NullHTTPRequest) && Controller::has_curr()) {
+            $request = Controller::curr()->getRequest();
+        }
+        
+        $stateStr = $this->getStateManager()->getStateFromRequest($this, $request);
+
+        if ($stateStr) {
+            $oldState = $this->getState(false);
+            // Create a dummy state so that we can merge the current state with the request state.
+            $newState = new GridState($this, $stateStr);
+            // Put the current state on top of the request state.
+            $newState->setValue($oldState->Value());
+            $this->state = $newState;
+        }
+    }
+
+    /**
+     * Add get and post parameters if exist to the URL.
+     * URL keeps state for GridField, that exist in few different tabs.
+     */
+    public function addOtherGridfieldStateToURL(string $link, bool $addState = true): string
+    {
+        $request = $this->getRequest();
+
+        if ($request->param('Action')) {
+            if (($request instanceof NullHTTPRequest) && Controller::has_curr()) {
+                $request = Controller::curr()->getRequest();
+            }
+    
+            $requestVars = $request->requestVars();
+            $postparams = [];
```

Comment:
```suggestion
            $params = [];
```
These aren't necessarily from `POST`.

Reply:
DONE